From 962ef491651039341583877e191ef920173d3f4c Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Tue, 14 Feb 2012 21:04:14 +0200
Subject: [PATCH 33/58] Ignore TX status for Data frames from not associated
 STA

Commit d9a38716ccf90b0ab307f570a30931684a1b8730 did this for
hostapd_eapol_tx_status() but missed the older hostapd_tx_status()
path. Address that case, too.

The TX status event may be received after a station has been
disassociated in cases where the disassociation is following a
transmission of a Data frame. Ignore such events if the STA is not
associated at the moment the event is being processed. This avoids
confusing debug entries and rescheduling of the EAPOL TX timeouts for
STAs that are still in the STA table, but are not really in active EAPOL
session.

Signed-hostap: Jouni Malinen <j@w1.fi>
intended-for: hostap-1
(cherry picked from commit 0c01d65d6deb4b4d721d255d0e7ca24d8ea2c6b5)
---
 src/ap/ieee802_11.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -1725,7 +1725,7 @@ void hostapd_tx_status(struct hostapd_da
 				break;
 		}
 	}
-	if (sta == NULL)
+	if (sta == NULL || !(sta->flags & WLAN_STA_ASSOC))
 		return;
 	if (sta->flags & WLAN_STA_PENDING_POLL) {
 		wpa_printf(MSG_DEBUG, "STA " MACSTR " %s pending "
